src/DotNetWorker.Core/Context/IBindingCache.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    internal interface IBindingCache<T>
```

Comment:
Feel free to suggest a better name for this type. It is a per invocation cache. `IInvocationDataCache` ?